### PR TITLE
25 BottomNavigation(Item) landscape support

### PR DESCRIPTION
**Pull request by alburdette619**
Added in orientation sensing and two modes for rendering landscape nav bar.  By default the items should keep their width and have the same widths.  If the `horizontalWhenLandscape/horizontal` prop is added, the labels and icons will be laid out horizontally and take up the full width of the bar both per Material Design.

Landscape Dos under Placement: https://material.io/design/components/bottom-navigation.html#placement

Also Storybook stories were added for the case of icon+label for horizontal mode, properties in docs were updated, Material Icons were linked in the ios project, and react-dom also here was added to dev dependencies.

TL;DR: Fix for #25 and add horizontal layout mode for landscape.

Screen Shot 2019-08-04 at 8 00 29 PM

Screen Shot 2019-08-04 at 8 00 51 PM

### CODE REVIEW

**Comment by codypearce** [comment]
@alburdette619 Wow these are some great changes, I was going to leave it up to the user to handle `Dimensions` but I think this is better. Also looks like Material changed to allow horizontal icon layouts since I made that issue, cool.

I think this is a case that would definitely benefit from a separate section in the docs to explain how to use these props. We could add another issue to handle if you'd like though.

**Comment by codypearce** [comment]
@alburdette619 Do you think you'll be able to add the new sections or would you like me?

**Comment by alburdette619** [comment]
@codypearce sorry, I haven't had a lot of extra time for this recently.  If you want to create another ticket or implement that here it's fine.  I've updated the branch from master.

**Comment by codypearce** [comment]
@alburdette619 I pushed some demos to the docs and storybook, thanks again for this awesome PR!
